src/helpers/email.service.ts

```diff
@@ -0,0 +1,88 @@
+import cron from 'node-cron'
+import nodemailer from 'nodemailer'
+import RateService from '../services/rate/rate.service'
+import UserService from '../services/user/user.service'
+import logger from './logger'
+import { EmailConfig, EmailDetails } from './interface/sendEmail.interface'
+import { User } from '../entity/user.entity'
+
+const emailConfig: EmailConfig = {
+    service: process.env.EMAIL_SERVICE || 'smtp.gmail.com',
+    login: process.env.EMAIL_LOGIN as string,
+    password: process.env.EMAIL_PASSWORD as string,
+    sender: process.env.EMAIL_SENDER as string,
+    subject: process.env.EMAIL_SUBJECT || 'USD to UAH Exchange Rate',
+    textTemplate: process.env.EMAIL_TEXT || '1 USD to UAH - {rate}',
+}
+
+const transporter = nodemailer.createTransport({
+    host: emailConfig.service,
+    port: 587,
+    secure: false,
+    auth: {
+        user: emailConfig.login,
+        pass: emailConfig.password,
+    },
+})
+
+async function sendEmails(): Promise<void> {
+    let users: User[], rate: number
+
+    try {
+        rate = await RateService.getExchangeRate()
```

Comment:
What will you do if you will need to need to use Monobank api to get the rate ?

Reply:
I should rewrite service in Factory pattern to provide different api

Reply:
Nice point, but I thought about Interface for this service and use it through constructor for the first time. In this case u can remove specific service usage in code and move specific declaration of it on the level above